token-details-persister/src/main/scala/it/pagopa/interop/tokendetailspersister/FileUtils.scala

```diff
@@ -2,32 +2,25 @@ package it.pagopa.interop.tokendetailspersister
 
 import com.typesafe.scalalogging.Logger
 import it.pagopa.interop.commons.files.service.FileManager
+import it.pagopa.interop.commons.files.service.impl.S3ManagerImpl
 import it.pagopa.interop.commons.utils.service.OffsetDateTimeSupplier
 
-import java.time.OffsetDateTime
-import java.time.format.DateTimeFormatter
-import java.util.UUID
 import scala.concurrent.Future
 
 final class FileUtils(val fileManager: FileManager, val dateTimeSupplier: OffsetDateTimeSupplier) {
 
-  private val logger: Logger         = Logger(this.getClass)
-  private val df: DateTimeFormatter  = DateTimeFormatter.ofPattern("yyyyMMdd")
-  private val dtf: DateTimeFormatter = DateTimeFormatter.ofPattern("yyyyMMdd_HHmmss")
-  private val cPath: String          = ApplicationConfiguration.containerPath.stripMargin('/')
-  private val fPath: String          = ApplicationConfiguration.tokenStoragePath.stripMargin('/')
-
-  private def fileName(now: OffsetDateTime): String = s"${now.format(dtf)}_${UUID.randomUUID()}.ndjson"
+  private val logger: Logger = Logger(this.getClass)
+  private val cPath: String  = ApplicationConfiguration.containerPath.stripMargin('/')
+  private val fPath: String  = ApplicationConfiguration.tokenStoragePath.stripMargin('/')
 
   def store(lines: List[String]): Future[String] = {
-    val now           = dateTimeSupplier.get()
-    val today         = now.format(df)
-    val fName: String = fileName(now)
-    logger.info(s"Storing ${lines.size} lines at $cPath/$fPath/$today/$fName")
+    val jwtPath: String = fileManager.asInstanceOf[S3ManagerImpl].createJWTPath()
```

Comment:
I think I should avoid this approach.
I suggest two ways to prevent this `asInstanceOf`:

- Put the definition of `createJWTPath` directly in the trait definition
- Move the function under the `utils` module

Reply:
> I think I should avoid this approach. I suggest two ways to prevent this `asInstanceOf`:
> 
>     * Put the definition of `createJWTPath` directly in the trait definition
> 
>     * Move the function under the `utils` module

I would prefer the first because makes sense that this function is located inside that module but putting the definition in the trait would require to implement it inside `FileManagerImpl` too; in this case how do you suggest to implement it there?